### PR TITLE
Add static mode check on data()

### PR DESCRIPTION
**Pull request by zhiqiu**
### PR types

Others
### PR changes

APIs
### Describe

As Paddle-2.0 set dynamic graph as default in #27443, we need to add some checks for `the static only` APIs. 
Typically, `paddle.static.data() / paddle.fluid.data() / paddle.fluid.layers.data()` needs to be checked since it is the entrance of static graph model.

```
>>> import paddle
>>> a = paddle.fluid.data('a', [2,3])
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "<decorator-gen-25>", line 2, in data
  File "/usr/local/lib/python3.5/dist-packages/paddle/fluid/wrapped_decorator.py", line 25, in __impl__
    return wrapped_func(*args, **kwargs)
  File "/usr/local/lib/python3.5/dist-packages/paddle/fluid/framework.py", line 229, in __impl__
    ), "We only support %s in static graph mode, please call 'paddle.enable_static()' to enter static graph mode." % func.__name__
AssertionError: We only support 'data()' in static graph mode, please call 'paddle.enable_static()' to enter static graph mode.
```

### CODE REVIEW

**Review by phlrain — DISMISSED** [review]

File: python/paddle/fluid/framework.py [hunk 1]
```diff
@@ -217,7 +217,16 @@ def __impl__(*args, **kwargs):
 def _dygraph_only_(func):
     def __impl__(*args, **kwargs):
         assert in_dygraph_mode(
-        ), "We Only support %s in dynamic mode, please call 'paddle.disable_static()' to enter dynamic mode." % func.__name__
+        ), "We only support %s in dynamic graph mode, please call 'paddle.disable_static()' to enter dynamic graph mode." % func.__name__
```
Comment (lanxianghit) [inline python/paddle/fluid/framework.py]:
```suggestion
        ), "We only support '%s' API in dynamic graph mode, please call 'paddle.disable_static()' to enter dynamic graph mode." % func.__name__
```

下同

Comment (zhiqiu) [reply python/paddle/fluid/framework.py]: 好的

**Review by lanxianghit — DISMISSED** [review]

**Review by luotao1 — APPROVED** [review]

**Review by XiaoguangHu01 — APPROVED** [review]
LGTM

File: python/paddle/fluid/framework.py (hunk 1 above)
Comment (python-bookworm) [reply python/paddle/fluid/framework.py]: 你是怎么解决这个问题的？我也遇到同样的问题，目前没有解决

**Pull request merged**